### PR TITLE
refactoring: remove project reference

### PR DESCRIPTION
**Pull request by surli**
Historically there was a ProjectReference to contain information about project like sources, classpath and tests classes and a singleton Config to contain information about timeouts, weight, and localizer/solver types. 

However with changes introduces by @tdurieux on patches and by @danglotb with nopol-server, Config is no longer a singleton and it contains also information on sources, classpath and test classes. 

Then I propose on that PR to simply remove ProjectReference and to only consider Config as the entry point of Nopol to repair a project.

### CODE REVIEW

File: nopol/src/main/java/fr/inria/lille/evo/Main.java [hunk 1]
```diff
@@ -70,7 +69,7 @@ public static void main(String[] args) throws IOException, ParseException {
         options.addOption("destSrcTestFolder", true, "java files where new tests will be added after validation. (ex: project/src/test) default: srcTestFolder");
         options.addOption("destCpTestFolder", true, "java files where new tests will be compiled after validation. (ex: project/target/test-classes) default: cpTestFolder");
         options.addOption("dependencies", true, "all other class or jar required for the news tests. cpClassFolde & cpTestFolder are not necessary. (ex: junit, hamcrest, evosuite)");
-        options.addOption("testClasses", true, "test classes used to generate patch (default : null = all classes)");
+        options.addOption("getTestClasses", true, "test classes used to generate patch (default : null = all classes)");
```
Comment (tdurieux) [inline nopol/src/main/java/fr/inria/lille/evo/Main.java]: why did you change the option name?

File: nopol/src/main/java/fr/inria/lille/evo/Main.java [hunk 2]
```diff
@@ -235,22 +236,20 @@ public static void main(String[] args) throws IOException, ParseException {
...
+        //create getClasspath
         //URL[] classPath = FileUtils.getURLs(sources.split(File.pathSeparator));
         URL[] classPath = JavaLibrary.classpathFrom(cp);
 
         logger.debug("Launch nopol with:");
         logger.debug("sources = "+sources);
-        logger.debug("classpath = "+cp);
-        logger.debug("testClasses = "+testClasses);
-
+        logger.debug("getClasspath = "+cp);
```
Comment (tdurieux) [inline nopol/src/main/java/fr/inria/lille/evo/Main.java]: why did you change these lines?

File: nopol/src/main/java/fr/inria/lille/repair/nopol/jpf/JPFUtil.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ public static Config createConfig(String[] args, String qualifiedNameClass, Stri
         conf.setProperty("search.heuristic.branch.no_branch_return", "-1");
 
         conf.setProperty("sourcepath", sourcePath);
-        conf.setProperty("classpath", classpath);
+        conf.setProperty("getClasspath", classpath);
```
Comment (tdurieux) [inline nopol/src/main/java/fr/inria/lille/repair/nopol/jpf/JPFUtil.java]: do not change this line

**Comment by tdurieux** [comment]
I propose to rename Config to NopolContext or something like this

File: nopol/src/main/java/fr/inria/lille/evo/Main.java (hunk 1 above)
Comment (surli) [reply nopol/src/main/java/fr/inria/lille/evo/Main.java]: argh. Wrong refactoring. I'll fix that.

File: nopol/src/main/java/fr/inria/lille/evo/Main.java (hunk 2 above)
Comment (surli) [reply nopol/src/main/java/fr/inria/lille/evo/Main.java]: Same as above.

**Comment by surli** [comment]
@monperrus ok to merge?
